sources/packages/api/src/route-controllers/education-program-offering/education-program-offering.controller.ts

```diff
@@ -72,4 +74,55 @@ export class EducationProgramOfferingController {
       offeringDelivered: offering.offeringDelivered,
     }));
   }
+
+  @HasLocationAccess("locationId")
+  @Get("location/:locationId/education-program/:programId/offering/:offeringId")
+  async getProgramOffering(
+    @Param("locationId") locationId: number,
+    @Param("programId") programId: number,
+    @Param("offeringId") offeringId: number,
+  ): Promise<ProgramOfferingDto> {
+    //To retrive Education program offering corresponding to ProgramId and LocationId
+    const offering = await this.programOfferingService.getProgramOffering(
+      locationId,
+      programId,
+      offeringId,
+    );
+    if (!offering) {
+      throw new UnprocessableEntityException(
+        "Not able to find a Education Program Offering associated with the current Education Program, Location and offering.",
+      );
+    }
+    return offering;
+  }
+
+  @HasLocationAccess("locationId")
+  @Patch(
```

Comment:
Open for discussion, maybe PUT would be appropriate because we are replacing basically the entire entity data.

Reply:
I believe, the difference is obvious and in API updates, I feel Patch works better than PUT as we do not know all the entities of the particular dto. I do agree with you that we are basically replacing the entire data, but Patch has a control of updating with Primary key of the DTO not just deleting the existing row and creating a new one altogether.

Reply:
After a conversation, we agreed to continue with the PATCH.
Besides that, I am not sure from what is the origin of the statement has origin: "_but Patch has a control of updating with Primary key of the DTO not just deleting the existing row and creating a new one altogether._". Can you point some reference with definition?

Reply:
https://en.wikipedia.org/wiki/Patch_verb#PUT_vs_PATCH_vs_POST